plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -352,6 +356,35 @@ def callback(table):
         self.create_bigquery_table(database, schema, table_name, callback,
                                    'CREATE DAY PARTITIONED TABLE')
 
+    def copy_bq_table(self, source, destination, materialization):
+        conn = self.get_thread_connection()
+        client = conn.handle
+
+        if materialization is 'incremental':
+            write_disposition = WRITE_APPEND
+        else:
+            write_disposition = WRITE_TRUNCATE
```

Comment:
This is a style/feelings thing, but it seems to me that the argument to `copy_bq_table` should be the write disposition, not the materialization. This if-block can go into the caller. Similarly, this code should take the results of those `table_ref` calls, not Relations. Generally, `connections.py` knows about the details of connections in some adapter-specific way, and `impl.py` presents a more relation-based interface.